shared-spring/src/main/java/tds/shared/spring/interceptors/RestTemplateLoggingInterceptor.java

```diff
@@ -34,18 +35,25 @@ public RestTemplateLoggingInterceptor(final ObjectMapper objectMapper) {
     public ClientHttpResponse intercept(final HttpRequest request,
                                         final byte[] body,
                                         final ClientHttpRequestExecution clientHttpRequestExecution) throws IOException {
-        final UUID traceId = UUID.randomUUID();
+
+        // get tracer id that used to associate all rest calls that are a child of this request
+        final String traceId = Optional.fromNullable(MDC.get(TRACER_ID_HEADER)).or(UUID.randomUUID().toString());
         logRequest(request, body, traceId);
-        ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
-        logResponse(response, traceId);
-        return response;
+        try {
+            ClientHttpResponse response = clientHttpRequestExecution.execute(request, body);
+            logResponse(response, traceId);
+            return response;
+        } catch (IOException e) {
+            log.error(String.format("Exception occurred while executing rest request. METHOD: %s URI: %s", request.getMethod(), request.getURI()), e);
+            throw e;
```

Comment:
I will update the PR so that logResponse will not change the control flow and the end user will still receive an error message from the rest execute method.